internal/pkg/aws/ecs/service.go

```diff
@@ -123,32 +123,66 @@ type ServiceArn string
 // For example: arn:aws:ecs:us-west-2:1234567890:service/my-project-test-Cluster-9F7Y0RLP60R7/my-project-test-myService-JSOH5GYBFAIB
 // will return my-project-test-Cluster-9F7Y0RLP60R7
 func (s *ServiceArn) ClusterName() (string, error) {
+	parsedArn, err := s.parse()
+	if err != nil {
+		return "", err
+	}
+	return parsedArn.clusterName, nil
+}
+
+type parsedServiceArn struct {
```

Comment:
i wonder if it makes sense to do something like:
```
type ServiceARN struct {
  // AccountID, Partition, Region, etc
}

func ParseServiceARN() (ServiceARN, error) {
  // contents of parse()
}
```

then users of the struct can pull fields without `parse()`ing multiple times